### DISCUSSION TITLE
How to Add an Itemstack NBT Tans

### DISCUSSION THREAD

**Discussion by IceBreathNetDevelopment** [Minestom/Minestom#985] [Help] [answered]
Hey i want to create a Banner and add them some NBT Tags, how i can do this ?

**Comment by TheMode** [comment, marked as answer]
You can use `ItemStack#fromNBT`
```java
ItemStack.fromNBT(Material.BLACK_BANNER, NBT.Compound(Map.of("key", NBT.String("value"))));
```

Or use the [Tag API](https://wiki.minestom.net/feature/tags) which work directly on NBT. (using `ItemStack#withTag`/`ItemMeta.Builder#set`)

**Reply by IceBreathNetDevelopment** [reply]
Thank you :)
